src/com.rs

```diff
@@ -100,3 +104,154 @@ impl<T> Hash for WeakPtr<T> {
         self.0.hash(state);
     }
 }
+
+/// Macro that allows generation of an easy to use enum for dealing with many different possible versions of a COM object.
+///
+/// Give the variants so that parents come before children. This often manifests as going up in order (1 -> 2 -> 3). This is vital for safety.
+///
+/// Three function names need to be attached to each variant. The examples are given for the MyComObject1 variant below:
+/// - the from function (`WeakPtr<actual::ComObject1> -> Self`)
+/// - the as function (`&self -> Option<WeakPtr<actual::ComObject1>>`)
+/// - the unwrap function (`&self -> WeakPtr<actual::ComObject1>` panicing on failure to cast)
+///
+/// ```no_compile,rust
```

Comment:
I can't find anything that indicates `no_compile` is an attribute that is available here. I used it in a small test crate and it doesn't seem to stop cargo from trying to compile it as a test.

This issue suggests that there is no way to do this but the most recent comment does contain a workaround:
https://github.com/rust-lang/rust/issues/63193

Reply:
My whole life is a lie!

That hack is pretty terrible though